src/Symfony/Component/Clock/TimeInterface.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Clock;
+
+/**
+ * @author Nicolas Grekas <p@tchwork.com>
+ */
+interface TimeInterface
+{
+    public function now(): \DateTimeImmutable;
+
+    public function sleep(float $seconds): void;
```

Comment:
`int|float`?

If an end-user class using strict types calls this API, could it be a problem?
(https://3v4l.org/mst1W, `sleep` would be called in context where strict types in not defined).
Implementations may need explicit cast, maybe some tests are missing?

Reply:
strict mode :facepalm:
But anyway, good idea, PR updated (but I won't add tests specific to this, testing the engine is not needed.)